### ISSUE TITLE
[Bug] ClaimsPrincipalExtensions.GetNameIdentifierId uses utid instead of sub claim

### ISSUE THREAD

**Issue by felickz** [AzureAD/microsoft-identity-web#171]
**Which Version of Microsoft Identity Web are you using ?**
Note that to get help, you need to run the latest version. 
 Microsoft Identity Web 0.1.3-preview 
https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs
**Where is the issue?**
* Web App
    * [x] Sign-in users
    * [ ] Sign-in users and call web APIs
* Web API
    * [ ] Protected web APIs (Validating tokens)
    * [ ] Protected web APIs (Validating scopes)
    * [ ] Protected web APIs call downstream web APIs
* Token cache serialization
     * [ ] In Memory caches
     * [ ] Session caches
     * [ ] Distributed caches

Other? - please describe;

**Is this a new or existing app?**

**Repro**

```csharp
        /// <summary>
        /// Gets the NameIdentifierId associated with the <see cref="ClaimsPrincipal"/>.
        /// </summary>
        /// <param name="claimsPrincipal">the <see cref="ClaimsPrincipal"/> from which to retrieve the sub claim.</param>
        /// <returns>Name identifier ID (sub) of the identity, or <c>null</c> if it cannot be found.</returns>
        public static string GetNameIdentifierId(this ClaimsPrincipal claimsPrincipal)
        {
            return claimsPrincipal.FindFirstValue(ClaimConstants.UniqueObjectIdentifier);
        }
```

**Expected behavior**
As per documentation - this should be looking for SUB claim?

**Actual behavior**
Uses 
```csharp
        /// <summary>
        /// UniqueObjectIdentifier: "utid".
        /// </summary>
        public const string UniqueObjectIdentifier = "utid";
```

**Possible Solution**
```csharp
            return claimsPrincipal.FindFirstValue(ClaimConstants.Sub);
```

**Comment by jmprieur** [comment]
@felickz: which documentation?

**Comment by jmprieur** [comment]
@felickz ?

**Comment by felickz** [comment]
@jmprieur  - the XML comment which indicates it pulls the "sub claim" - which is published to docs: https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.web.claimsprincipalextensions.getnameidentifierid?view=azure-dotnet-preview#parameters

> Parameters

**ClaimsPrincipal**
the [ClaimsPrincipal ](https://docs.microsoft.com/en-us/dotnet/api/system.security.claims.claimsprincipal) from which to retrieve the sub claim.

**Comment by felickz** [comment]
Only stumbled on this because our b2c integration code fell into a odd flow, if this would have been looking at the sub claim as documented - my code would have worked out of the box.  I cannot find any documentation on what the **utid** is intended to be used- some internal MSAL claim?

**Comment by jmprieur** [comment]
Thanks @felickz for your explanations

**Comment by pmaytak** [comment]
Fixed in [Microsoft Identity Web 0.1.5-preview release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/0.1.5-preview).

cc: @felickz
